config.md

```diff
@@ -265,6 +265,19 @@ Annotations are key-value maps.
 }
 ```
 
+## ImagePath
+
+ImagePath is an optional absolute path to the root filesystem image used by VM-based runtimes.
+
+## Kernel-configuration
+
+Used by VM-based runtimes only.
+
+* **`path`** (string, required) specifies an absolute path to the kernel
+  used to boot the virtual machine.
+* **`parameters`** (string, optional) specifies a space-separated list of parameters to pass to the kernel.
+* **`initrd`** (string, optional) specifies an absolute path to an initial ramdisk to be used by the kernel.
```

Comment:
On Mon, May 16, 2016 at 06:35:07AM -0700, James Hunt wrote:
> - it's cwd.

Ccon uses it's cwd (e.g. see wording in [1]), and that approach hasn't
given me any trouble.

> - will not be available below `$bundle_dir`.

Why force this restriction at the runtime level?  I'm fine if you want
to enforce it at a higher level for your implementation, but I see no
technical reason to keep the kernel binary outside the bundle
directory (and of course, the absolute path could lead you back into
the bundle directory, just not in a very portable manner).

> On that topic, I note that all hook paths must be absolute.

Yeah, I'm also not a fan of that restriction ;).

> Also, absolute paths are more in line with the imperative nature of
> the configuration file - don't give the runtime choices (unless
> there's are good reason to ;-)

If they're relative to the runtime's cwd (or use runtime PATH lookups
for hooks), then there are still no *runtime* choices.  It just makes
some workflows more convenient for the bundle author and runtime
caller (e.g. you can bundle your kernel instead of transmitting it
out-of-band).

> Since paths can be troublesome, I wonder if all paths mentioned by
> the spec should be reviewed to ensure that for each path that is
> permitted to be relative, a rationale be provided as to why it can
> be relative?

I think this is a policy decision that is probably useful to do at a
higher level, but I'd rather keep policy out of the runtime spec.